lib/deref.js

```diff
@@ -84,36 +84,36 @@ module.exports = {
   /**
    * Creates a schema that's a union of all input schemas (only type: object is supported)
    *
-   * @param {array} schemaArr - array of schemas, all of which must be valid in the returned object
-   * @param {string} parameterSourceOption tells that the schema object is of request or response
-   * @param {*} components components in openapi spec.
-   * @param {object} schemaResolutionCache stores already resolved references
-   * @param {*} resolveFor - resolve refs for validation/conversion (value to be one of VALIDATION/CONVERSION)
-   * @param {string} resolveTo The desired JSON-generation mechanism (schema: prefer using the JSONschema to
+   * @param {array} schemaArr REQUIRED - array of schemas, all of which must be valid in the returned object
+   * @param {string} parameterSourceOption REQUIRED tells that the schema object is of request or response
+   * @param {*} components REQUIRED components in openapi spec.
+   * @param {object} options - REQUIRED a list of options to indicate the type of resolution needed.
+   * @param {*} options.resolveFor - resolve refs for validation/conversion (value to be one of VALIDATION/CONVERSION)
+   * @param {string} options.resolveTo The desired JSON-generation mechanism (schema: prefer using the JSONschema to
      generate a fake object, example: use specified examples as-is). Default: schema
-   * @param {*} stack counter which keeps a tab on nested schemas
-   * @param {*} seenRef References that are repeated. Used to identify circular references.
-   * @param {*} stackLimit Depth to which the schema should be resolved.
+   * @param {*} options.stack counter which keeps a tab on nested schemas
+   * @param {*} options.seenRef References that are repeated. Used to identify circular references.
+   * @param {*} options.stackLimit Depth to which the schema should be resolved.
    * @returns {*} schema - schema that adheres to all individual schemas in schemaArr
    */
-  resolveAllOf: function (schemaArr, parameterSourceOption, components, schemaResolutionCache,
-    resolveFor = 'CONVERSION', resolveTo = 'schema', stack = 0, seenRef, stackLimit) {
+  resolveAllOf: function (schemaArr, parameterSourceOption, components,
+    { resolveFor = 'CONVERSION', resolveTo = 'schema', stack = 0, seenRef = {}, stackLimit = 10, analytics = {} }) {
```

Comment:
Possible improvement: We can have single source of truth for some of these values. i.e. default stacklimit can be picked up from this single source of truth instead of using 10 here.

Reply:
Yes. Makes sense. Let me make that change.